### PR TITLE
Allow Forge create table ENUM and SET with array constraint

### PR DESCRIPTION
**Pull request by natanfelles**
See: https://forum.codeigniter.com/thread-61466.html

### CODE REVIEW

**Comment by natanfelles** [comment]
Continues running like:

```php
$db_columns = array(
    'status' => array(
        'type'    => "ENUM('unlocked','locked')",
        'default' => 'unlocked',
        'comment' => 'Status of asset (Locked or Unlocked)'
    )
); 
```

And now with type and separated array constraint:

```php
$db_columns = array(
    'status' => array(
        'type'       => 'ENUM',
        'constraint' => ['unlocked', 'locked'],
        'default'    => 'unlocked',
        'comment'    => 'Status of asset (Locked or Unlocked)'
    )
); 
```

File: tests/system/Database/Live/ForgeTest.php [hunk 1]
```diff
@@ -280,4 +280,43 @@ public function testDropForeignKey()
 		$this->forge->dropTable('forge_test_users', true);
 
 	}
+
+	public function testEnumSetFields()
+	{
+		if ($this->db->DBDriver !== 'MySQLi')
```
Comment (lonnieezell) [inline tests/system/Database/Live/ForgeTest.php]: We need to test Postgres, also. And SQLite is coming fairly soon....

Comment (natanfelles) [reply tests/system/Database/Live/ForgeTest.php]: As I understand, not all databases have `ENUM` data type and `SET` is more specific in MySQL.

In the case of SQLite, it does not have any of these two types.

I thought about separating the test of each type into methods and then go updating the list of valid databases as they arrive.

**Comment by lonnieezell** [comment]
I was thinking Postgres had SET - but it's a totally different feature. However, it [does have ENUM](https://www.postgresql.org/docs/9.3/static/datatype-enum.html) support in all of the currently supported versions. 

According to [this SO post](https://stackoverflow.com/questions/5299267/how-to-create-enum-type-in-sqlite#5301032) there's a way to build a custom ENUM solution, but definitely not for our needs. 

So - please add the test for Postgres ENUM, and make sure the docs state which platforms the commands are available on.

**Comment by lonnieezell** [comment]
I know it's been a while. Sorry. Getting back into the thick of things.

Looks like there's 1 failing test that needs fixing up. Can you tackle that @natanfelles ?

> 1) CodeIgniter\Database\Live\ForgeTest::testEnumFields
> Failed asserting that null matches expected 'enum_string'.
> /home/travis/build/bcit-ci/CodeIgniter4/tests/system/Database/Live/ForgeTest.php:306

**Comment by natanfelles** [comment]
Hello! I did not know how to proceed with postgres.

**Comment by natanfelles** [comment]
@lonnieezell 

Postgres uses ENUM to "CREATE TYPE" - so, CREATE TABLE uses the "created type", not ENUM.

https://www.postgresql.org/docs/9.3/datatype-enum.html 

We would need to use an extra query to create each type and run them before CREATE TABLE. I do not think that fits.

**Pull request closed without merging**